src/gas_price.rs

```diff
@@ -76,6 +79,17 @@ impl EstimatedGasPrice {
             eip1559: self.eip1559.map(|x| x.limit_cap(cap)),
         }
     }
+
+    pub fn is_valid(&self) -> bool {
+        self.cap() >= self.tip() && self.cap() >= self.base_fee()
+    }
+
+    pub fn valid(self) -> Result<EstimatedGasPrice> {
```

Comment:
Maybe better name is `validate`.
I would also like to have an explanation on is_valid or valid where this criteria comes from.

Reply:
fixed